modules/reactor/src/meshgenerators/FlexiblePatternGenerator.C

```diff
@@ -553,9 +564,23 @@ FlexiblePatternGenerator::FlexiblePatternGenerator(const InputParameters & param
 std::unique_ptr<MeshBase>
 FlexiblePatternGenerator::generate()
 {
-  if (isParamValid("external_boundary_id"))
-    MooseMesh::changeBoundaryId(
-        **_build_mesh, OUTER_SIDESET_ID, getParam<boundary_id_type>("external_boundary_id"), false);
+  MooseMesh::changeBoundaryId(**_build_mesh, OUTER_SIDESET_ID, _external_boundary_id, false);
```

Comment:
```suggestion
  if (_external_boundary_id != OUTER_SIDESET_ID)
    MooseMesh::changeBoundaryId(**_build_mesh, OUTER_SIDESET_ID, _external_boundary_id, false);
```
Would still be helpful though it is fine to re-id a boundary using the same id.